chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -59,6 +59,150 @@ trait ChainUnitTest
+        extends FixtureTag("BitcoindZmqChainHandlerWithBlock")
+
+    def from(tag: String): FixtureTag = {
+      tag match {
+        case Empty.name                   => Empty
+        case GenisisBlockHeaderDAO.name   => GenisisBlockHeaderDAO
+        case PopulatedBlockHeaderDAO.name => PopulatedBlockHeaderDAO
+        case GenisisChainHandler.name     => GenisisChainHandler
+        case BitcoindZmqChainHandlerWithBlock.name =>
+          BitcoindZmqChainHandlerWithBlock
+        case _: String =>
+          throw new IllegalArgumentException(s"$tag is not a valid tag")
+      }
+    }
+  }
+
+  def defaultTag: FixtureTag = FixtureTag.Empty
+
+  sealed trait ChainFixture
+
+  object ChainFixture {
+    case object Empty extends ChainFixture
+
+    case class GenisisBlockHeaderDAO(dao: BlockHeaderDAO) extends ChainFixture
+
+    case class PopulatedBlockHeaderDAO(dao: BlockHeaderDAO) extends ChainFixture
+
+    case class GenisisChainHandler(chainHandler: ChainHandler)
+        extends ChainFixture
+
+    case class BitcoindZmqChainHandlerWithBlock(
+        bitcoindChainHandler: BitcoindChainHandler)
+        extends ChainFixture
+
+    def create(tag: FixtureTag): Future[ChainFixture] = {
+      tag match {
+        case FixtureTag.Empty => Future.successful(ChainFixture.Empty)
+        case FixtureTag.GenisisBlockHeaderDAO =>
+          createBlockHeaderDAO().map(GenisisBlockHeaderDAO.apply)
+        case FixtureTag.PopulatedBlockHeaderDAO =>
+          createPopulatedBlockHeaderDAO().map(PopulatedBlockHeaderDAO.apply)
+        case FixtureTag.GenisisChainHandler =>
+          createChainHandler().map(GenisisChainHandler.apply)
+        case FixtureTag.BitcoindZmqChainHandlerWithBlock =>
+          createBitcoindChainHandler().map(
+            BitcoindZmqChainHandlerWithBlock.apply)
+      }
+    }
+
+    def destroy(fixture: ChainFixture): Future[Any] = {
+      fixture match {
+        case Empty                      => Future.successful(())
+        case GenisisBlockHeaderDAO(_)   => destroyHeaderTable()
+        case PopulatedBlockHeaderDAO(_) => destroyHeaderTable()
+        case GenisisChainHandler(_)     => destroyHeaderTable()
+        case BitcoindZmqChainHandlerWithBlock(bitcoindHandler) =>
+          destroyBitcoindChainHandler(bitcoindHandler)
+      }
+    }
+  }
+
+  def withChainFixture(test: OneArgAsyncTest): FutureOutcome = {
+    val stringTag = test.tags.headOption.getOrElse(defaultTag.name)
+
+    val fixtureTag: FixtureTag = FixtureTag.from(stringTag)
+
+    val fixtureF: Future[ChainFixture] = ChainFixture.create(fixtureTag)
+
+    val outcomeF = fixtureF.flatMap(fixture =>
+      test(fixture.asInstanceOf[FixtureParam]).toFuture)
+
+    val fixtureTakeDownF = outcomeF.flatMap { outcome =>
+      val destroyedF =
+        fixtureF.flatMap(fixture => ChainFixture.destroy(fixture))
+
+      destroyedF.map(_ => outcome)
+    }
+
+    new FutureOutcome(fixtureTakeDownF)
+  }
+
+  final class SugaryItVerbStringTaggedAs(
```

Comment:
Can we get a comment on the methods and on this class? I have no idea what it is meant to do. This seems to be where the test case actually gets run?